vortex-btrblocks/src/integer/stats.rs

```diff
@@ -205,13 +210,14 @@ where
         } else {
             HashMap::with_hasher(FxBuildHasher)
         },
-        distinct_values_count: if count_distinct_values { 0 } else { u32::MAX },
         prev: head,
         runs: 1,
     };
 
-    let values = array.buffer::<T>();
-    let mask = validity.to_boolean_buffer();
+    let values = buffer.slice(head_idx..array.len());
```

Comment:
you could also just advance offset below to head_idx, then avoid the extra slice

Reply:
not quite sure, we are reallying on chunking the buffer slice. Could convert it into chunking the iterator but thought this is nicer

Reply:
hmm, let me try something

Reply:
I think it’s still easiest to slice but I’m reworking this code to use array_chunks like we do in bitpacking